src/features/processPicker.ts

```diff
@@ -246,4 +333,16 @@ function execChildProcess(process: string, workingDirectory: string): Promise<st
             resolve(stdout);
         });
     });
+}
+
+function execChildProcessWithTimeout(process: string, workingDirectory: string): Promise<string> {
+    return new Promise<string>((resolve, reject) => {
+        child_process.exec(process, { timeout: 3000, cwd: workingDirectory, maxBuffer: 500 * 1024 }, (error: Error, stdout: string, stderr: string) => {
+            if (stderr && stderr.length > 0) {
+                reject(stderr);
```

Comment:
We may well need to prepend this with some text for this to be comprehensible. Something like 'Pipe transport program bla failed.\n\n<message>'